walkthrough/06-includes-vs-stacks/04-combining-patterns/README.md

```diff
@@ -0,0 +1,164 @@
+
+Additionally, note the introduction of the `environment.yml` file in the environment stacks. Many teams today today use non-HCL configurations in conjunction with HCL configurations to define configurations for their environments. The `values` attribute of the `terragrunt.stack.hcl` file is useful, and many teams might prefer it to an `environment.hcl` file found in a parent folder. However, if you are a team that is already using configuration files like `environment.yml`, and have additional tooling like `yq` that interacts with those files, you might prefer to continue using those files. This is a good example of how the Stacks pattern really aims to be additive to your existing workflows and patterns, rather than a complete replacement. You don't have to rewrite all of your configurations if it doesn't benefit you.
+
+## Conclusion
+
+Hopefully, these examples give you a good idea of how you can use the different patterns enabled by Terragrunt to shape configurations in a way that works for your team, and the trade-offs they present.
+
+The "Not DRY" pattern is a good way to get started with Terragrunt, and is a good way to get a feel for how Terragrunt works. It's also a good way to get started with Terragrunt if you're not sure how you want to structure your configurations yet. However, as your infrastructure grows, you'll likely find that you want to share more configuration between units, and the "Not DRY" pattern can become unwieldy.
+
+The "_envcommon" pattern is a good way to quickly configuration between units without larger refactoring, and is a valid way to get started with reusing configurations. However, as the number of units you manage grows, you'll likely find that you want to share more configuration between environments, and the "_envcommon" pattern can become unwieldy.
+
+The "Stacks" pattern is a good way to manage configurations at scale, and is a good way to manage configurations across environments. However, it requires a different way of thinking about how you structure your configurations, and might not be a good fit for all teams, especially ones that have already invested heavily in the "_envcommon" pattern.
+
+To be clear, I'm also not saying that the `_envcommon` pattern is a definite stepping stone to the "Stacks" pattern. Many teams have successfully managed infrastructure at large scale using the "_envcommon" pattern, and if it works for your team, then by all means, use it. Similarly, if you want to start from a scalable foundation, and you know that you will encounter the problems that the "_envcommon" pattern presents, then the "Stacks" pattern might be a good place to start after getting started with Terragrunt.
```

Comment:
My instinct is we should have a clear best practice for 95% of users, and call out that "some teams at scale may see value in deviating from this practice by doing XYZ for ABC reason, but for most teams Terragrunt is intended to be used as XYZ"